cpp/src/c_api/capi_helper.cu

```diff
@@ -74,6 +75,28 @@ template void sort_by_key(raft::handle_t const& handle,
                           raft::device_span<int64_t> keys,
                           raft::device_span<int64_t> values);
 
+template <typename T>
```

Comment:
I would suggest moving this to `src/detail`.  It has applicability outside of the C API, this is exposing the device comms methods to an `.cpp` file.

Reply:
File would need a new name, perhaps `collect_comm_wrapper.cu`.

Reply:
I moved this functionality to `src/detail` per your recommendation.